src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1009,9 +1025,20 @@ def enable_model_cpu_offload(self, gpu_id: Optional[int] = None, device: Union[t
         hook = None
         for model_str in self.model_cpu_offload_seq.split("->"):
             model = all_model_components.pop(model_str, None)
+            is_loaded_in_4bit_bnb, is_loaded_in_8bit_bnb = False, False
+            if model is not None and isinstance(model, torch.nn.Module):
+                _, is_loaded_in_4bit_bnb, is_loaded_in_8bit_bnb = _check_bnb_status(model)
+
             if not isinstance(model, torch.nn.Module):
                 continue
 
+            # This is because the model would already be placed on a CUDA device.
+            if is_loaded_in_8bit_bnb:  # is_loaded_in_4bit_bnb or is_loaded_in_8bit_bnb:
+                logger.info(
+                    f"Skipping the hook placement for the {model.__class__.__name__} as it is loaded in `bitsandbytes` 8bit."
+                )
+                continue
+
```

Comment:
we should also skip for 4-bit when `.to()` is not supported

Reply:
Not sure if I understand. I have `bitsandbytes` to be installed with at least 0.43.3:

https://github.com/huggingface/diffusers/blob/859f2d76bdccf28e3ae35660c60cd9df946b197c/src/diffusers/quantizers/bitsandbytes/bnb_quantizer.py#L76

So, I guess 4bit model offloading here should be possible?

Reply:
Oh nice then !